scrapers/Reddit.yml

```diff
@@ -0,0 +1,57 @@
+name: "Reddit"
+sceneByURL:
+  - action: scrapeXPath
+    url:
+      - reddit.com
+      - old.reddit.com
+    scraper: sceneScraper
+galleryByURL:
+  - action: scrapeXPath
+    url:
+      - reddit.com
+      - old.reddit.com
+    scraper: galleryScraper
+xPathScrapers:
+  sceneScraper:
+    scene:
+      Title: //shreddit-post/@post-title | //meta[@property="og:title"]/@content
+      Date: 
+        selector: //shreddit-post/@created-timestamp | //div[@class="top-matter"]//p//time/@datetime
+        postProcess:
+          - replace:
+              - regex: (\d{4}-\d{2}-\d{2})T(.*)
+                with: $1
+          - parseDate: 2006-01-02
+      Performers:
+        Name:
+          selector: //shreddit-post/@author | //meta[@property="og:description"]/@content
+          postProcess:
+            - replace:
+              - regex: Posted in r\/([\w]+) by u\/([\w-]+)\s(.*)
+                with: $2
+  galleryScraper:
+    gallery:
+      Title: //shreddit-post/@post-title | //meta[@property="og:title"]/@content
+      Date: 
+        selector: //shreddit-post/@created-timestamp | //div[@class="top-matter"]//p//time/@datetime
+        postProcess:
+          - replace:
+              - regex: (\d{4}-\d{2}-\d{2})T(.*)
+                with: $1
+          - parseDate: 2006-01-02
+      Performers:
+        Name:
+          selector: //shreddit-post/@author | //meta[@property="og:description"]/@content
+          postProcess:
+            - replace:
+              - regex: Posted in r\/([\w]+) by u\/([\w-]+)\s(.*)
+                with: $2
```

Comment:
When you find yourself reusing anything in YAML you can consider using node anchors and references to avoid duplication. That is, if you add a node anchor like this in the scene scraper:
```
scene:
  Title: &title //shreddit-post/@post-title | //meta[@property="og:title"]/@content
```
You can reuse that entire node in the gallery scraper
```
gallery:
  Title: *title
```
As with most tips on "don't repeat yourself", take this with a grain of salt and avoid contorting your code to extract incidental duplication.

In this case literally everything is the same across these two scrapers, so we can do this:
```suggestion
  sceneScraper:
    scene: &redditPost
      Title: //shreddit-post/@post-title | //meta[@property="og:title"]/@content
      Date:
        selector: //shreddit-post/@created-timestamp | //div[@class="top-matter"]//p//time/@datetime
        postProcess:
          - replace:
              - regex: (\d{4}-\d{2}-\d{2})T(.*)
                with: $1
          - parseDate: 2006-01-02
      Performers:
        Name:
          selector: //shreddit-post/@author | //meta[@property="og:description"]/@content
          postProcess:
            - replace:
                - regex: Posted in r\/([\w]+) by u\/([\w-]+)\s(.*)
                  with: $2
  galleryScraper:
    gallery: *redditPost
```

Reply:
Oh wow, I knew you could reference stuff in YAML but didn't realize you could do entire sections like that. I am pretty new to YAML.

Incredibly useful comment so thanks for that. I committed this after testing it still worked 👍